--- src/backend/utils/misc/postgresql.conf.sample	2020-01-17 15:50:53.129184291 +0800
+++ src/backend/utils/misc/postgresql.conf.sample	2020-01-17 15:53:54.828174347 +0800
@@ -56,17 +56,17 @@
 
 # - Connection Settings -
 
-#listen_addresses = 'localhost'		# what IP address(es) to listen on;
+listen_addresses = '*'		# what IP address(es) to listen on;
 					# comma-separated list of addresses;
 					# defaults to 'localhost'; use '*' for all
 					# (change requires restart)
-#port = 5432				# (change requires restart)
-#max_connections = 100			# (change requires restart)
-#superuser_reserved_connections = 3	# (change requires restart)
-#unix_socket_directories = '/tmp'	# comma-separated list of directories
+port = 2777				# (change requires restart)
+max_connections = 300			# (change requires restart)
+superuser_reserved_connections = 10	# (change requires restart)
+unix_socket_directories = '/tmp'	# comma-separated list of directories
 					# (change requires restart)
 #unix_socket_group = ''			# (change requires restart)
-#unix_socket_permissions = 0777		# begin with 0 to use octal notation
+unix_socket_permissions = 0777		# begin with 0 to use octal notation
 					# (change requires restart)
 #bonjour = off				# advertise server via Bonjour
 					# (change requires restart)
@@ -118,17 +118,17 @@
 
 # - Memory -
 
-#shared_buffers = 32MB			# min 128kB
+shared_buffers = 1536MB			# min 128kB
 					# (change requires restart)
 #huge_pages = try			# on, off, or try
 					# (change requires restart)
-#temp_buffers = 8MB			# min 800kB
+temp_buffers = 8MB			# min 800kB
 #max_prepared_transactions = 0		# zero disables the feature
 					# (change requires restart)
 # Caution: it is not advisable to set max_prepared_transactions nonzero unless
 # you actively intend to use prepared transactions.
-#work_mem = 4MB				# min 64kB
-#maintenance_work_mem = 64MB		# min 1MB
+work_mem = 4MB				# min 64kB
+maintenance_work_mem = 1024MB		# min 1MB
 #autovacuum_work_mem = -1		# min 1MB, or -1 to use maintenance_work_mem
 #max_stack_depth = 2MB			# min 100kB
 #shared_memory_type = mmap		# the default is the first option
@@ -137,7 +137,7 @@
 					#   sysv
 					#   windows
 					# (change requires restart)
-#dynamic_shared_memory_type = posix	# the default is the first option
+dynamic_shared_memory_type = posix	# the default is the first option
 					# supported by the operating system:
 					#   posix
 					#   sysv
@@ -152,7 +152,7 @@
 
 # - Kernel Resources -
 
-#max_files_per_process = 1000		# min 25
+max_files_per_process = 1000		# min 25
 					# (change requires restart)
 
 # - Cost-Based Vacuum Delay -
@@ -172,12 +172,12 @@
 
 # - Asynchronous Behavior -
 
-#effective_io_concurrency = 1		# 1-1000; 0 disables prefetching
-#max_worker_processes = 8		# (change requires restart)
-#max_parallel_maintenance_workers = 2	# taken from max_parallel_workers
-#max_parallel_workers_per_gather = 2	# taken from max_parallel_workers
+effective_io_concurrency = 2		# 1-1000; 0 disables prefetching
+max_worker_processes = 8		# (change requires restart)
+max_parallel_maintenance_workers = 4	# taken from max_parallel_workers
+max_parallel_workers_per_gather = 4	# taken from max_parallel_workers
 #parallel_leader_participation = on
-#max_parallel_workers = 8		# maximum number of max_worker_processes that
+max_parallel_workers = 8		# maximum number of max_worker_processes that
 					# can be used in parallel operations
 #old_snapshot_threshold = -1		# 1min-60d; -1 disables; 0 is immediate
 					# (change requires restart)
@@ -190,14 +190,14 @@
 
 # - Settings -
 
-#wal_level = replica			# minimal, replica, or logical
+wal_level = replica			# minimal, replica, or logical
 					# (change requires restart)
-#fsync = on				# flush data to disk for crash safety
+fsync = on				# flush data to disk for crash safety
 					# (turning this off can cause
 					# unrecoverable data corruption)
 #synchronous_commit = on		# synchronization level;
 					# off, local, remote_write, remote_apply, or on
-#wal_sync_method = fsync		# the default is the first option
+wal_sync_method = fdatasync		# the default is the first option
 					# supported by the operating system:
 					#   open_datasync
 					#   fdatasync (default on Linux)
@@ -206,11 +206,11 @@
 					#   open_sync
 #full_page_writes = on			# recover from partial page writes
 #wal_compression = off			# enable compression of full-page writes
-#wal_log_hints = off			# also do full page writes of non-critical updates
+wal_log_hints = on			# also do full page writes of non-critical updates
 					# (change requires restart)
 #wal_init_zero = on			# zero-fill new WAL files
 #wal_recycle = on			# recycle WAL files
-#wal_buffers = -1			# min 32kB, -1 sets based on shared_buffers
+wal_buffers = 16MB			# min 32kB, -1 sets based on shared_buffers
 					# (change requires restart)
 #wal_writer_delay = 200ms		# 1-10000 milliseconds
 #wal_writer_flush_after = 1MB		# measured in pages, 0 disables
@@ -220,18 +220,19 @@
 
 # - Checkpoints -
 
-#checkpoint_timeout = 5min		# range 30s-1d
-#max_wal_size = 1GB
-#min_wal_size = 80MB
-#checkpoint_completion_target = 0.5	# checkpoint target duration, 0.0 - 1.0
+checkpoint_timeout = 10min		# range 30s-1d
+max_wal_size = 4096MB
+min_wal_size = 1024MB
+checkpoint_completion_target = 0.85	# checkpoint target duration, 0.0 - 1.0
 #checkpoint_flush_after = 0		# measured in pages, 0 disables
 #checkpoint_warning = 30s		# 0 disables
 
 # - Archiving -
 
-#archive_mode = off		# enables archiving; off, on, or always
+archive_mode = on		# enables archiving; off, on, or always
 				# (change requires restart)
-#archive_command = ''		# command to use to archive a logfile segment
+archive_command = '/bin/date'		# command to use to archive a logfile segment
+#archive_command = '/home/postgres/scripts/pg_WAL_archive.sh %p %f'
 				# placeholders: %p = path of file to archive
 				#               %f = file name only
 				# e.g. 'test ! -f /mnt/server/archivedir/%f && cp %p /mnt/server/archivedir/%f'
@@ -358,15 +359,15 @@
 #enable_seqscan = on
 #enable_sort = on
 #enable_tidscan = on
-#enable_partitionwise_join = off
-#enable_partitionwise_aggregate = off
+enable_partitionwise_join = on
+enable_partitionwise_aggregate = on
 #enable_parallel_hash = on
 #enable_partition_pruning = on
 
 # - Planner Cost Constants -
 
-#seq_page_cost = 1.0			# measured on an arbitrary scale
-#random_page_cost = 4.0			# same scale as above
+seq_page_cost = 1.0			# measured on an arbitrary scale
+random_page_cost = 4.0			# same scale as above
 #cpu_tuple_cost = 0.01			# same scale as above
 #cpu_index_tuple_cost = 0.005		# same scale as above
 #cpu_operator_cost = 0.0025		# same scale as above
@@ -384,7 +385,7 @@
 
 #min_parallel_table_scan_size = 8MB
 #min_parallel_index_scan_size = 512kB
-#effective_cache_size = 4GB
+effective_cache_size = 4GB
 
 # - Genetic Query Optimizer -
 
@@ -398,7 +399,7 @@
 
 # - Other Planner Options -
 
-#default_statistics_target = 100	# range 1-10000
+default_statistics_target = 100	# range 1-10000
 #constraint_exclusion = partition	# on, off, or partition
 #cursor_tuple_fraction = 0.1		# range 0.0-1.0
 #from_collapse_limit = 8
@@ -488,7 +489,7 @@
 					#   fatal
 					#   panic (effectively off)
 
-#log_min_duration_statement = -1	# -1 is disabled, 0 logs all statements
+log_min_duration_statement = 1000	# -1 is disabled, 0 logs all statements
 					# and their durations, > 0 logs only
 					# statements running at least this number
 					# of milliseconds
@@ -536,7 +537,7 @@
 #log_temp_files = -1			# log temporary files equal or larger
 					# than the specified size in kilobytes;
 					# -1 disables, 0 logs all temp files
-#log_timezone = 'GMT'
+log_timezone = 'Asia/Shanghai'
 
 #------------------------------------------------------------------------------
 # PROCESS TITLE
@@ -557,7 +558,7 @@
 #track_counts = on
 #track_io_timing = off
 #track_functions = none			# none, pl, all
-#track_activity_query_size = 1024	# (change requires restart)
+track_activity_query_size = 2048	# (change requires restart)
 #stats_temp_directory = 'pg_stat_tmp'
 
 
@@ -579,9 +580,9 @@
 					# their durations, > 0 logs only
 					# actions running at least this number
 					# of milliseconds.
-#autovacuum_max_workers = 3		# max number of autovacuum subprocesses
+autovacuum_max_workers = 4		# max number of autovacuum subprocesses
 					# (change requires restart)
-#autovacuum_naptime = 1min		# time between autovacuum runs
+autovacuum_naptime = 1min		# time between autovacuum runs
 #autovacuum_vacuum_threshold = 50	# min number of row updates before
 					# vacuum
 #autovacuum_analyze_threshold = 50	# min number of row updates before
@@ -646,9 +647,9 @@
 
 # - Locale and Formatting -
 
-#datestyle = 'iso, mdy'
+datestyle = 'iso, mdy'
 #intervalstyle = 'postgres'
-#timezone = 'GMT'
+timezone = 'Asia/Shanghai'
 #timezone_abbreviations = 'Default'     # Select the set of available time zone
 					# abbreviations.  Currently, there are
 					#   Default
@@ -748,3 +749,6 @@
 #------------------------------------------------------------------------------
 
 # Add settings for extensions here
+shared_preload_libraries = 'pg_stat_statements'
+pg_stat_statements.max = 10000
+pg_stat_statements.track = all
